src/oca_pre_commit_hooks/checks_odoo_module_xml.py

```diff
@@ -348,3 +348,18 @@ def check_xml_xpath(self):
                         f'{manifest_data["filename_short"]}:{xpath_node.sourceline} '
                         f"Use of translatable xpath `text()`"
                     )
+
+    @utils.only_required_for_checks("xml-oe-structure-missing-id")
+    def check_xml_oe_structure(self):
+        """* Check xml-oe-structure-missing-id
+
+        Ensure all tags with class 'oe_structure' have an ID. For more information on the rationale see:
+        https://github.com/OCA/odoo-pre-commit-hooks/issues/27
+        """
+        for manifest_data in self.manifest_datas:
+            for xpath_node in manifest_data["node"].xpath("//*[contains(@class, 'oe_structure')]"):
```

Comment:
Does `hasclass` work outside Odoo? It did not work when I tested it out. However I used a different xpath to prevent the false positives you (I think) refer to:

This previous implementation would match stuff like `random_oe_structure`, or `oe_structure_suffix`, however that should not be the case now. I added test cases to verify it.

Reply:
:O

I wasn't aware `hasclass()` was not a native xpath function.

But anyways, you can do the same as [Odoo does](https://github.com/odoo/odoo/blob/27ba86323ff5d57887a6901ffc997ba17a559262/odoo/addons/base/models/ir_ui_view.py#L220).

For more info, check https://github.com/odoo/odoo/commit/9cefa76988ff94c3d590c6631b604755114d0297

Reply:
Only this comment is missing. I consider it's worth it to apply it, because code is less clearer otherwise

Reply:
Done, I changed it to the Odoo way :handshake: